Assets/Mirror/Tests/SyncDictionaryTest.cs

```diff
@@ -0,0 +1,234 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace Mirror.Tests
+{
+    [TestFixture]
+    public class SyncDictionaryTest
+    {
+        public class SyncDictionaryIntString : SyncDictionary<int, string> {}
+
+        SyncDictionaryIntString serverSyncDictionary;
+        SyncDictionaryIntString clientSyncDictionary;
+
+        private void SerializeAllTo<T>(T fromList, T toList) where T: SyncObject
+        {
+            NetworkWriter writer = new NetworkWriter();
+            fromList.OnSerializeAll(writer);
+            NetworkReader reader = new NetworkReader(writer.ToArray());
+            toList.OnDeserializeAll(reader);
+        }
+
+        private void SerializeDeltaTo<T>(T fromList, T toList) where T : SyncObject
+        {
+            NetworkWriter writer = new NetworkWriter();
+            fromList.OnSerializeDelta(writer);
+            NetworkReader reader = new NetworkReader(writer.ToArray());
+            toList.OnDeserializeDelta(reader);
+            fromList.Flush();
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            serverSyncDictionary = new SyncDictionaryIntString();
+            clientSyncDictionary = new SyncDictionaryIntString();
+
+            // add some data to the list
+            serverSyncDictionary.Add(0, "Hello");
+            serverSyncDictionary.Add(1, "World");
+            serverSyncDictionary.Add(2, "!");
```

Comment:
what happens when we use 'null' string as value?

Reply:
there are test cases that cover the null case.    it would add a 2 => null  entry in the dictionary.  This is working as expected

Reply:
As Paul said, there are 2 tests for null set. I thought maybe you meant null key, but int cannot be null.